### PR TITLE
AYS-368 | Role Listing Service Automation

### PR DESCRIPTION
**Pull request by cagla88**
## Pull Request

**Description:**
Role Listing Service Automation

**Related Issue:**
Closes #[AYS-368]

**Changes Made:**
Role listing positive tests added without filtering
Role listing positive scenarios have been added using all filtering data.
Added new user which name is TEST ADMIN
Role listing negative scenarios have been added.

**Additional Information:**
Add any additional information that might be relevant to reviewers.

## Checklist
- [x] I have read and followed the [contribution guidelines](README.md#contributing) for this project.
- [x] I have added tests (if applicable).
- [x] All tests are passing.
- [x] I have updated the documentation (if necessary).

## Reviewer Instructions
Provide instructions or guidelines for reviewers to follow when assessing your pull request.

[AYS-368]: https://afetyonetimsistemi.atlassian.net/browse/AYS-368?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/test/java/org/ays/tests/institution/roleManagementService/PostRolesTest.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        volunteerFoundation.testVolunteerFoundationRoleCount();
+        int totalElementCount = volunteerFoundation.getDbUserCount();
+        VolunteerFoundation.tearDown();
+
+        Response response = InstitutionEndpoints.listRoles(rolesListPayload);
+
+        if (response.jsonPath().getList("response.content").isEmpty()) {
+            AysLogUtil.info("No users under this institution.");
+            return;
+        }
```
Comment (Nbhtozcn) [inline src/test/java/org/ays/tests/institution/roleManagementService/PostRolesTest.java:42]:
```
if (response.jsonPath().getList("response.content").isEmpty()) {
            AysLogUtil.info("No roles under this institution.");
            return;
        }
```

File: src/test/java/org/ays/tests/institution/roleManagementService/PostRolesTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        disasterFoundation.testDisasterFoundationRoleCount();
+        int totalElementCount = disasterFoundation.getDbUserCount();
+        DisasterFoundation.tearDown();
+
+        Response response = InstitutionEndpoints.listRolesForAdminTwo(rolesListPayload);
+
+        if (response.jsonPath().getList("response.content").isEmpty()) {
+            AysLogUtil.info("No users under this institution.");
+            return;
+        }
```
Comment (Nbhtozcn) [inline src/test/java/org/ays/tests/institution/roleManagementService/PostRolesTest.java:68]: if (response.jsonPath().getList("response.content").isEmpty()) {
            AysLogUtil.info("No roles under this institution.");
            return;
        }

File: src/test/java/org/ays/tests/institution/roleManagementService/PostRolesTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        afetYonetimSistemi.testAYSRoleCount();
+        int totalElementCount = afetYonetimSistemi.getDbUserCount();
+        AfetYonetimSistemi.tearDown();
+
+        Response response = InstitutionEndpoints.listRolesForSuperAdmin(rolesListPayload);
+
+        if (response.jsonPath().getList("response.content").isEmpty()) {
+            AysLogUtil.info("No users under this institution.");
+            return;
+        }
```
Comment (Nbhtozcn) [inline src/test/java/org/ays/tests/institution/roleManagementService/PostRolesTest.java:94]:
```
if (response.jsonPath().getList("response.content").isEmpty()) {
            AysLogUtil.info("No roles under this institution.");
            return;
        }
```

Comment (cagla88) [reply src/test/java/org/ays/tests/institution/roleManagementService/PostRolesTest.java]: @Nbhtozcn Selamlar, istediğiniz tüm değişiklikleri yapıp gönderdim tekrar review edebilir misiniz?

**Review by Nbhtozcn — APPROVED** [review]

**Pull request merged**